virttest/utils_misc.py

```diff
@@ -1611,30 +1617,144 @@ def get_cpu_info(session=None):
+        :param pkg: pacakge name or list
+        """
+        if not isinstance(session, (aexpect.ShellSession, aexpect.Expect)):
+            raise exceptions.TestError("Parameters exception on session")
+        if not isinstance(pkg, list):
+            if not isinstance(pkg, str):
+                raise exceptions.TestError("pkg %s must be list or str" % pkg)
+            else:
+                self.pkg_list = [pkg, ]
+        else:
+            self.pkg_list = pkg
+        self.package_manager = None
+        self.query_cmd = None
+        self.install_cmd = None
+        self.remove_cmd = None
+        self.session = session
+
+        # Inspect and set package commands
+        for pkg_mgr in PACKAGE_MANAGERS:
+            cmd = 'which ' + pkg_mgr
+            if not self.session.cmd_status(cmd):
+                self.package_manager = pkg_mgr
+                break
+
+            if not self.package_manager:
+                raise exceptions.TestError("Package manager not in %s" %
+                                           PACKAGE_MANAGERS)
+
+        if self.package_manager == 'apt-get':
+            self.query_cmd = "dpkg -s "
+            self.remove_cmd = "apt-get --purge remove -y "
+            self.install_cmd = "apt-get install -n "
+        else:
+            self.query_cmd = "rpm -q "
+            self.remove_cmd = self.package_manager + " remove -y "
+            self.install_cmd = self.package_manager + " install -y "
+
+    def is_installed(self, pkg_name):
+        """
+        Check the package install status
+
+        :param pkg_name: package name
+        :return: True or False
+        """
+        cmd = self.query_cmd + pkg_name
+        return not self.session.cmd_status(cmd)
+
+    def install(self, pkg_list, timeout=300):
+        """
+        Use package manager install packages
+
+        :param pkg_list: package list
+        :return: if install succeed return True, else False
+        """
+        for pkg in pkg_list:
+            cmd = self.install_cmd + pkg
+            if (('*' in pkg or not self.is_installed(pkg)) and
+                    self.session.cmd_status(cmd, timeout)):
+                logging.error("Install package %s failed", pkg)
+                return False
+        return True
+
+    def remove(self, pkg_list, timeout=300):
+        """
+        Use package manager remove packages
 
-    :param pkg_list: list of packages
-    :session: session Object
+        :param pkg_list: list of packages
+        :param timeout: remove timeout
+        :return: if remove succeed return True, else False
+        """
+        for pkg in pkg_list:
+            cmd = self.remove_cmd + pkg
+            if (("*" in pkg or self.is_installed(pkg)) and
+                    self.session.cmd_status(cmd, timeout)):
+                logging.error("Remove package %s failed", pkg)
+                return False
+        return True
+
+
+def package_install(pkg, session=None, timeout=300):
```

Comment:
just a suggestion, both `package_install` and `package_remove` are almost same instead we can have a single function, 
```
def package_manager(pkg, session=None, func=sm_inst.install, timeout=300):
    for pkg in pkg_list:
        if (('*' in pkg or not sm_inst.check_installed(pkg)) and
            not func(pkg)):
```